### PR TITLE
feat: sidebar fixes

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Introduces a new sidebar structure and flow.
I decided to do some major cleanup as there where many pieces we didn't use or where legacy.
Also was hard to follow rendering so did some structure cleanup.

For now I left the old sidebar parts as Nimrod is debating to ruin experiment or not:
https://dailydotdev.slack.com/archives/C02AA18RD1D/p1730443529887579

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-665 #done

### CODE REVIEW

File: packages/shared/src/components/sidebar/sections/ResourceSection.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        target: '_blank',
+      },
+      {
+        icon: () => <ListIcon Icon={() => <TerminalIcon />} />,
+        title: 'Changelog',
+        path: `${webappUrl}sources/daily_updates`,
+        rightIcon: () =>
+          changelog.isAvailable && (
+            <AlertDot className="-top-1 right-0" color={AlertColor.Cabbage} />
+          ),
```
Comment (omBratteng) [inline packages/shared/src/components/sidebar/sections/ResourceSection.tsx:31]: Do we still use this?

Comment (rebelchris) [reply packages/shared/src/components/sidebar/sections/ResourceSection.tsx:31]: Not 100% sure, just leaving it for now for a second

Comment (omBratteng) [reply packages/shared/src/components/sidebar/sections/ResourceSection.tsx:31]: 👍

**Review by omBratteng — APPROVED** [review]
LGTM

File: packages/shared/src/components/MainLayout.tsx [hunk 2]
```diff
@@ -46,9 +46,10 @@ const GoBackHeaderMobile = dynamic(
   { ssr: false },
 );
 
-const Sidebar = dynamic(
-  () => import(/* webpackChunkName: "sidebar" */ './sidebar/Sidebar'),
-  { ssr: false },
```
Comment (sshanzel) [inline packages/shared/src/components/MainLayout.tsx:51]: Why did we end up removing the ssr: false?

**Review by sshanzel — APPROVED** [review]
Great to see so much cleanups! 💪

File: packages/shared/src/components/MainLayout.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/MainLayout.tsx:51]: Doesn't feel like we need it actually here?
But maybe you or @capJavert know the replication case I could check to break it without SSR: false?
